### PR TITLE
Make lists be comparable

### PR DESCRIPTION
**Pull request by Infinisil**
Makes lists comparable using lexicographic comparison.

Increments builtins.langVersion in order for this change to be
detectable

Implements https://github.com/NixOS/nix/issues/5627

### CODE REVIEW

**Comment by Infinisil** [comment]
Actually, there's two different ways of doing lexicographic ordering, and I'm not sure which one makes more sense:
1. Smaller lists are always earlier, with strings: "a", "b", "ab", "bb"
2. The first letter decides the order, with strings: "a", "ab", "b", "bb"

This PR currently implements 1., but I think 2. makes more sense, partly because it's also how strings are ordered in Nix, and also the second one is what I need for my use-case.

**Comment by Infinisil** [comment]
Also 2 is used by Haskell

**Comment by edolstra** [comment]
I think lexicographic (i.e. dictionary) ordering almost always has meaning 2. For instance, it's what `operator <` on `std::vector` does in C++. The only place I've seen meaning 1 is in the ordering of keys in canonical CBOR...

**Comment by Infinisil** [comment]
I updated this PR to use number 2 instead. Apparently both versions can be called lexicographic order, see [Wikipedia](https://en.wikipedia.org/wiki/Lexicographic_order), but number 2 is the usual one.

**Comment by edolstra** [comment]
Can you add this to the release notes and possibly update the docs for the `<` operator?

File: src/libexpr/primops.cc [hunk 1]
```diff
@@ -3103,7 +3119,7 @@ static void prim_lessThan(EvalState & state, const Pos & pos, Value * * args, Va
 {
     state.forceValue(*args[0], pos);
     state.forceValue(*args[1], pos);
-    CompareValues comp;
+    CompareValues comp = CompareValues(state);
```
Comment (edolstra) [inline src/libexpr/primops.cc:3122]:
```suggestion
    CompareValues comp{state};
```

**Comment by Infinisil** [comment]
@edolstra Added a release note, expanded the description of the `<` to mention lexicographic comparison, and applied your suggestions

**Pull request merged**